pwiz/data/msdata/MSData.hpp

```diff
@@ -439,6 +439,13 @@ struct PWIZ_API_DECL MZIntensityPair
 
 PWIZ_API_DECL std::ostream& operator<<(std::ostream& os, const MZIntensityPair& mzi);
 
+struct PWIZ_API_DECL NoiseDataInfo
+{
+    double mz;
+    float baseline;
+    float intensity;
+};
+
```

Comment:
I think I understand what you're saying. I was just copying the example of MZIntensityPair. Are you suggesting that I use 3 specific vectors (noise m/z, noise intensity, noise baseline) instead? Perhaps it would be even better if SpectrumList_Thermo.cpp called a generic "setBinaryArray" method on MSData 3 times instead. If other readers ever want to respect the --addNoiseData flag, they may want to set a different number or types of binary arrays for noise.

Reply:
I think a setBinaryDataArray is a good idea. Also swapBinaryDataArray. And yes, do that 3 times with 3 `vector<double>`s.